spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -219,3 +219,40 @@ Jakarta Data implementations often support the following list of predicate keywo
 |findByNameOrderByAgeAscNameDescYearAsc
 
 |===
+
+===== Special parameter handling
+
+Jakarta Data also supports particular parameters to define pagination and sorting.
+
+Jakarta Data recognizes, when specified on a repository method after the query parameters, specific types, like ```Pageable``` and ```Sort```, to dynamically apply pagination and sorting to queries. The following example demonstrates these features:
+
+[source,java]
+----
+@Repository
+public interface ProductRepository extends CrudRepository<Product, Long> {
+
+  List<Product> findByName(String name, Pageable pageable);
+
+  List<Product> findByName(String name, Sort sort);
+
+}
+----
+
+You can define simple sorting expressions by using property names.
+
+[source,java]
+----
+Sort name = Sort.asc("name");
+Sort nameAge = Sort.desc("name").add(Sort.asc("age"));
+----
+
+You can combine sorting with start and size by using property names.
+
+[source,java]
+----
+Pageable pageable = Pageable.of(1, 20);
+Pageable pageableSort = Pageable.of(1, 20, Sort.asc("name"));
+first20 = products.findByNameLike(name,
+                     Pageable.of(1, 20, Order.desc("price")));
```

Comment:
This example was intended to decouple the ordering/sorting from the pagination,

```
first20 = products.findByNameLike(name,
                     Pageable.of(1, 20),
                     Order.desc("price"));
```

Being able to specify order/sorting within a Pageable seems like bad practice that will encourage users to specify order/sorting dynamically when they don't really need it dynamic.  Having order/sorting be dynamic prevents preparing the JPQL in advance because ORDER BY is part of the query that gets prepared in the database.  It would be best to have the programming model encourage static specification of sorting, with dynamic only as a special case for when it's truly needed.